core/genesis.go

```diff
@@ -312,11 +312,11 @@ func SetupGenesisBlockWithOverride(db ethdb.Database, triedb *trie.Database, gen
 
 			if config.IsOptimism() && config.ChainID != nil && config.ChainID.Cmp(big.NewInt(params.OPGoerliChainID)) == 0 {
 				// Apply Optimism Goerli regolith time
-				config.RegolithTime = &params.OptimismGoerliRegolithTime
+				config.RegolithTime = superchain.OPChains[params.OPGoerliChainID].RegolithTime
 			}
 			if config.IsOptimism() && config.ChainID != nil && config.ChainID.Cmp(big.NewInt(params.BaseGoerliChainID)) == 0 {
 				// Apply Base Goerli regolith time
-				config.RegolithTime = &params.BaseGoerliRegolithTime
+				config.RegolithTime = superchain.OPChains[params.BaseGoerliChainID].RegolithTime
 			}
```

Comment:
Actually, let's not change anything about Regolith and leave all code untouched that sets Regolith timestamps, because we will very soon remove all Goerli-based chains, which are the only chains left that didn't have Regolith at genesis. (Base and OP Labs Goerli prod and devnet, and chaos net)

That means, we also shouldn't introduce the `regolith_time` field in https://github.com/ethereum-optimism/superchain-registry/pull/101